node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -213,39 +213,35 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
   }
 
   /** Broadcasts the given transaction over the P2P network */
-  override def broadcastTransaction(transaction: Transaction): Future[Unit] = {
-    val broadcastTx = BroadcastAbleTransaction(transaction)
+  override def broadcastTransactions(
+      transactions: Vector[Transaction]): Future[Unit] = {
+    val broadcastTxDbs = transactions.map(tx => BroadcastAbleTransaction(tx))
 
-    val addToDbF = txDAO.upsert(broadcastTx)
+    val addToDbF = txDAO.upsertAll(broadcastTxDbs)
```

Comment:
Hmm... so it seems like something funny is going here: 

Whose responsibility is to re-broadcast transactions? 

It seems that the wallet should tell the node that a transaction needs to be re-broadcast, but the node actually executes it. 

It seems like the node project should have an API that allows the wallet to tell it when 

1. We need to broadcast a transaction
2. We don't need to broadcast this transaction anymore (presumably because it is confirmed)

Now the node project itself should have all details about how often to broadcast, what peer to broadcast to (in the future), etc. 

I think the biggest limitation in the node project right now is we have no ability to do (2). 

Thoughts?

Reply:
This is where things get complicated. If we do this, then the node's db needs to be chain aware and the node will need to process incoming blocks and filters. imo it makes more sense for the wallet to handle all of this.

Reply:
AFIAK this is how core does it as well.

Reply:
So here are my thoughts: 

1. A new transaction originated from the wallet should be broadcast via a wallet callback
2. A wallet transaction confirmed (or pending confirmed, if we introduce a new TxoState) should also have a wallet callback. This callback can then _delist_ the transaction to be broadcast in the node project since it is confirmed.